cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -435,11 +436,12 @@ private void httpExceptionHandler(Promise<? extends CruiseControlResponse> resul
             // Vert.x throws a NoStackTraceTimeoutException (inherits from TimeoutException) when the request times out
             // so we catch and raise a TimeoutException instead
             result.fail(new TimeoutException(t.getMessage()));
-        } else if (t instanceof ConnectException) {
+        } else if (t instanceof NoRouteToHostException || t instanceof ConnectException) {
+            // Netty throws a AnnotatedNoRouteToHostException (inherits from NoRouteToHostException) and  when it cannot resolve the host
             // Vert.x throws a AnnotatedConnectException (inherits from ConnectException) when the request times out
-            // so we catch and raise a ConnectException instead
-            result.fail(new ConnectException(t.getMessage()));
-        } else {
+            // so we catch and raise a CruiseControlRetriableConnectException instead since
+            result.fail(new CruiseControlRetriableConnectException(t.getMessage()));
```

Comment:
Should this inherit the exception instead of copying only the message? That way, it should keep the whole stack etc. and I do not see what problems would it cause.

```suggestion
            result.fail(new CruiseControlRetriableConnectException(t));
```

Reply:
I tried doing this but I was getting some issues since the message which is being set in the condition seems to come out as `null` if I do that so probably up in the stack the message we are passing is being used

Reply:
But reusing the exception is important since that gives you the stack trace etc. I do not understand how the message being null relates to it.